File: frontend/src/objects/admin/shop/request/CreateShopItemRequest.ts
import { z } from "zod";

export const CreateShopItemRequestBodySchema = z.object({
  name: z.string().min(2),
  description: z.string().min(4),
  price: z.number().int().min(1),
  currency: z.enum(["coins", "gems"]),
  catalogIndex: z.number().int().min(0).default(0),
  active: z.boolean().default(true),
  sortOrder: z.number().int().default(0),
});

export type CreateShopItemRequestBody = z.infer<typeof CreateShopItemRequestBodySchema>;

File: backend/microservice/src/admin/objects/shop/request/CreateShopItemRequest.scala
package microservice.admin.objects.shop.request

import cats.effect.IO
import io.circe.generic.semiauto._
import io.circe.{Decoder, Encoder}
import org.http4s.EntityDecoder
import org.http4s.circe.jsonOf

/** POST /admin/shop/items 请求对象；字段语义见 [[microservice.player.objects.shop.ShopItem]]。 */
final case class CreateShopItemRequest(
  name: String,
  description: String,
  price: Int,
  currency: String,
  catalogIndex: Int = 0,
  active: Boolean = true,
  sortOrder: Int = 0
)

private[admin] object CreateShopItemRequest {
  implicit val encoder: Encoder[CreateShopItemRequest] = deriveEncoder
  implicit val decoder: Decoder[CreateShopItemRequest] = deriveDecoder
  implicit val entityDecoder: EntityDecoder[IO, CreateShopItemRequest] = jsonOf
}
